cpp/daal/src/algorithms/k_nearest_neighbors/bf_knn_impl.i

```diff
@@ -246,6 +251,7 @@ protected:
             int * indices = indexesBlock.get();
 
             const size_t size = blockSize * k * sizeof(*indices);
+            DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, blockSize * sizeof(*indices), k);
```

Comment:
I think you can do this check before the for.

Reply:
Fixed